### PR TITLE
ENH: Bump ITK and move to reusable workflow CI

### LINKED ISSUE #21: Update to `ITKRemoteModuleBuildTestPackageAction` reusable CI workflow

### PR DESCRIPTION
**Pull request by tbirdso**
Bumps ITK and replaces cookiecutter yml with reusable CI workflow.

Let's release a package for v5.3.0 compatibility before moving to the latest pre-release.

Closes https://github.com/InsightSoftwareConsortium/ITKGrowCut/issues/21

### CODE REVIEW

**Comment by tbirdso** [comment]
Seeing a C++ build error: https://open.cdash.org/build/9151198/configure
```
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
CMake Error at /home/runner/work/ITKGrowCut/ITK/CMake/ITKModuleAPI.cmake:78 (message):
  No such module: "ITKSmoothing"
Call Stack (most recent call first):
  /home/runner/work/ITKGrowCut/ITK/CMake/ITKModuleAPI.cmake:7 (itk_module_load)
  /home/runner/work/ITKGrowCut/ITK/CMake/ITKModuleAPI.cmake:153 (_itk_module_use_recurse)
  /home/runner/work/ITKGrowCut/ITK/CMake/ITKModuleMacros.cmake:324 (itk_module_use)
  test/CMakeLists.txt:1 (itk_module_test)


-- Configuring incomplete, errors occurred!
See also "/home/runner/work/ITKGrowCut/build/CMakeFiles/CMakeOutput.log".
```

File: .github/workflows/build-test-package.yml [hunk 1]
```diff
@@ -3,279 +3,14 @@ name: Build, test, package
...
-    - name: Publish 🐍 Python 📦 package to PyPI
-      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
-      uses: pypa/gh-action-pypi-publish@master
-      with:
-        user: __token__
-        password: ${{ secrets.pypi_password }}
+  cxx-build-workflow:
+    uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-cxx.yml@246883ef3828fc00219145aeec9efa67b9889d0b
+    with:
+      itk-cmake-options: '-DITK_BUILD_DEFAULT_MODULES:BOOL=OFF -DITKGroup_Core:BOOL=ON'
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml]:
```suggestion
      itk-cmake-options: '-DITK_BUILD_DEFAULT_MODULES:BOOL=OFF -DITKGroup_Core:BOOL=ON -DModule_ITKAntiAlias:BOOL=ON -DModule_ITKImageGrid:BOOL=ON -DModule_ITKSpatialObjects:BOOL=ON -DModule_ITKTestKernel:BOOL=ON -DModule_ITKMetaIO:BOOL=ON'
```

Comment (dzenanz) [reply .github/workflows/build-test-package.yml]: Add dependencies as per https://github.com/InsightSoftwareConsortium/ITKGrowCut/blob/master/itk-module.cmake

Comment (tbirdso) [reply .github/workflows/build-test-package.yml]: Great! Thanks for taking the lead on this.

**Comment by tbirdso** [comment]
Notebook failure is expected, I advise we move forward since other tests are green and I'll fix up that one CI line in a followup PR.

**Comment by dzenanz** [comment]
From the log:
```log
Run pytest --nbmake --nbmake-timeout=30000 examples/*.ipynb
ERROR: file or directory not found: examples/*.ipynb
```
I guess we should disable notebook tests, as there are none in this repo.

File: .github/workflows/build-test-package.yml [hunk 2]
```diff
@@ -3,279 +3,14 @@ name: Build, test, package
...
+
+  python-build-workflow:
+    uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@246883ef3828fc00219145aeec9efa67b9889d0b
+    with:
+      test-notebooks: true
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml]:
```suggestion
      test-notebooks: false
```

**Comment by dzenanz** [comment]
It looks like we were writing at the same time 😞

**Comment by dzenanz** [comment]
I would prefer if the last 2 commits were squashed into the second one.

**Comment by tbirdso** [comment]
> I would prefer if the last 2 commits were squashed into the second one.

Sounds good, feel free to make that change now or at merge. Would you like to wait for CI to complete or move forward with merge?

**Comment by dzenanz** [comment]
I think we can proceed with the merge. Could you squash the last 3 commits into one, and merge?

**Comment by tbirdso** [comment]
Squashed and merging following successful C++/Python builds: https://github.com/InsightSoftwareConsortium/ITKGrowCut/actions/runs/6878963343

**Pull request merged**